src/harness/reference_models/iap/iap.py

```diff
@@ -0,0 +1,551 @@
+    channels: List of tuples containing (low_freq, high_freq) of each 5MHz 
+              segment of the whole frequency range of protected entity
+    low_freq: low frequency of protection constraint (Hz)
+    high_freq: high frequency of protection constraint (Hz)
+    grant_objects: a list of grant requests, each one being a tuple 
+                   containing grant information
+    fss_info: contains fss point and antenna information
+    esc_antenna_info: contains information on ESC antenna height, azimuth, 
+                      gain and pattern gain
+    threshold: protection threshold (mW)
+    protection_ent_type: an enum member of class ProtectedEntityType 
+    asas_interference: object of class type AggregateInterferenceOutputFormat 
+    aggregate_interference: object of class type AggregateInterferenceOutputFormat 
+  Returns:
+    Updates asas_interference instance of class type AggregateInterferenceOutputFormat.
+    It is updated in the format of {latitude : {longitude : [ asas_p(mW/IAPBW),.. ]}}
+    and aggregate_interference instance of class type AggregateInterferenceOutputFormat.
+    It is updated in the format of {latitude : { longitude : [a_p(mW/IAPBW),..] }}
+  """
+  
+  # Get all the grants inside neighborhood of the protection entity
+  grants_inside = interf.findGrantsInsideNeighborhood(
+                    grant_objects, protection_point, protection_ent_type)
+    
+  if len(grants_inside) > 0:
+
+    for channel in channels:
+      # Set interference quota for protection_point, channel combination
+
+      # Calculate Roll of Attenuation for ESC Sensor Protection entity
+      if protection_ent_type is interf.ProtectedEntityType.ESC:
+        if channel[0] >= 3650.e6:
+          center_freq = (channel[0] + channel[1]) / 2
+          interference_quota = interf.dbToLinear(interf.linearToDb(threshold) - 
+                (2.5 + ((center_freq - interf.ESC_CH21_CF_HZ) / interf.MHZ)))
+        else:
+          interference_quota = threshold
+
+      # Get the FSS blocking passband range
+      elif protection_ent_type is interf.ProtectedEntityType.FSS_BLOCKING:
+        interference_quota = threshold
+        channel = [low_freq, high_freq]
+
+      else:
+        interference_quota = threshold
+
+      # Get protection constraint over 5MHz channel range
+      channel_constraint = interf.ProtectionConstraint(latitude=protection_point[1],
+                             longitude=protection_point[0], low_frequency=channel[0],
+                             high_frequency=channel[1], entity_type=protection_ent_type)
+
+      # Identify CBSD grants overlapping with frequency range of the protection entity 
+      # in the neighborhood of the protection point and channel
+      neighborhood_grants = interf.findOverlappingGrants(
+                              grants_inside, channel_constraint)
+      num_unsatisfied_grants = len(neighborhood_grants)
+
+      if not neighborhood_grants:
+        continue
```

Comment:
I'm afraid that we cannot simply "continue" here. Even if the protected point does not have neighborhood_grants, we still need to update asas_interference & aggregate_interference, because we have to evenly distribute the remaining interference_quota across all SASes.

Reply:
Considering   no grants are overlapping in the 5MHz channel, asas_interference and   aggregate_interference will be updated with 0    for that protection point , channel